### ISSUE TITLE
Installing Rancher (Single Cluster (rancherd))

### ISSUE THREAD

**Issue by maggieliu** [rancher/rancher#28131]
* Installation available on RancherD CLI
** Please refer to the internal RancherD design doc on Confluence for details

**Comment by StrongMonkey** [comment]
Here is the detailed implementation of rancherd:

Rancherd is a single binary released along with rancher. Master build of rancherd can be downloaded through https://releases.rancher.com/rancherd/rancherd-master. Rancherd is essentially rke2 + rancherd chart.

[Rancherd chart](https://github.com/rancher/rancher/tree/master/cmd/rancherd/chart) is a simplified version of rancher server chart we have today. 

-  rancher is deployed through **daemonset** instead of deployment in master node. This makes scaling rancher easier as it scales when you add more master nodes.

- rancher is exposed through hostport. This avoids applying ingress and ingress controller on top of rancher. To provide outbound access an external LB has to be configured to connect to hostports on server nodes. 

- cert-manager is no longer used to provision certs for rancher. In order to bring your own certs, the certs have to be provided on the following location on serve nodes. Rancherd chart will automatically mount these as hostPath into rancher pods.

 ```
 Private key: /etc/rancher/ssl/key.pem
 Certificate: /etc/rancher/ssl/cert.pem
 CA Certificate(self-signed): /etc/rancher/ssl/cacerts.pem
 Additional CA Certificate: /etc/ssl/certs/ca-additional.pem
```

- rancherd chart is customized through [helmChartConfig](https://github.com/rancher/rancher/tree/master/cmd/rancherd/chart#customize-chart) instead of overriding values.yaml in helm command. 

### Security improvement

Rancherd deploys rancher in a way so that no default admin will be generated on start-up, and in order to bootstrap default admin and password `rancherd reset-admin` has to be run on one of the server nodes. `rancherd reset-admin` provides the ability to bootstrap/reset default admin user and a random generated password. It will also print out a URL that redirects to UI to ask user to change password.(Note: this part has to be implemented in UI)

### QA

Rancherd can be installed throught package script that @dweomer is working on. Once rancherd is installed, run `rancherd reset-admin` to bootstrap admin and password. It should print out a URL, username and password. Click on the URL and fill out username and password and rancher should be able to use. Be sure to test the following use cases:

1. Make sure rancher is deployed to only master node. Rancher is also scalable with the number of master node in the cluster
2. Make sure rancher can be configured to access by an external LB that connects master node through 8080 and 8443.
3. Make sure different certs option still work by providing on disks. The workflow is similar with single docker install https://rancher.com/docs/rancher/v2.x/en/installation/other-installation-methods/single-node-docker/. Checkout certs option. if your certs is signed by a known CA, make sure to pass `publicCA: true` in [helmChartConfig](https://github.com/rancher/rancher/tree/master/cmd/rancherd/chart#customize-chart) as you have to customize the chart.
4. Make sure there is no default admin generated until `rancherd reset-admin`. Also test if `rancherd reset-admin` is able to bootstrap and reset password as intended.

**Comment by sangeethah** [comment]
Blocked on https://github.com/rancher/rancher/issues/29051

**Comment by MonzElmasry** [comment]
@sangeethah issue should be resolved now, This is ready to be tested.  https://github.com/rancher/rancher/issues/29051

**Comment by rancher-max** [comment]
#### Validated on v2.5.0-rc3

- **Issues found during this testing are tracked separately**
- Validated self-signed certs, rancher generated certs, and valid certs signed by a ca. When using options other than the rancher generated certs (which is the default option), you must pass `tls-san` value when bringing up the server. Ensure the cert covers the value passed for that. Otherwise, the cluster will not start correctly.
- Validated using helm config values. Very important when bringing your own certs to use publicCA.
- Using reset-admin functions correctly, although generally gives the private IP. It should just be known that the value passed in the tls-san should also have the server accessible on port 8443, or the node public ip on port 8443
- General functionality of downstream clusters appears to be working. Potential issue with ClusterIP on workloads but still debugging this and if so will open a separate issue.
- Rancher Dashboard works
